docs/csharp/programming-guide/main-and-command-args/top-level-statements.md

```diff
@@ -0,0 +1,90 @@
+---
+title: "Top-level statements - C# Programming Guide"
+description: Learn about top-level statements in C# 9 and later.
+ms.date: 03/08/2021
+helpviewer_keywords:
+  - "C# language, top-level statements"
+  - "C# language, Main method"
+---
+# Top-level statements (C# Programming Guide)
+
+Starting in C# 9, you don't have to explicitly include a `Main` method in a console application project. Instead, you can use the *top-level statements* feature to minimize the code you have to write. In this case, the compiler implicitly provides a class and `Main` method entry point for the application.
+
+Here's a *Program.cs* file that is a complete C# program in C# 9:
+
+:::code language="csharp" source="snippets/top-level-statements-1/Program.cs":::
+
+Top-level statements let you write simple programs for small utilities such as Azure Functions and GitHub Actions. They also make it simpler for new C# programmers to get started learning and writing code.
+
+The following sections explain the rules on what you can and can't do with top-level statements.
+
+## Only one top-level file
+
+An application must have only one entry point, so a project can have only one file with top-level statements. Putting top-level statements in more than one file in a project results in the following compiler error:
+
+> CS8802 Only one compilation unit can have top-level statements.
+
+A project can have any number of additional source code files that don't have top-level statements.
+
+## No other entry points
+
+You can write a `Main` method explicitly, but it can't function as an entry point. The compiler issues the following warning:
+
+> CS7022 The entry point of the program is global code; ignoring 'Main()' entry point.
+
+In a project with top-level statements, you can't use the [-main](../../language-reference/compiler-options/main-compiler-option.md) compiler option to select the entry point, even if the project has one or more `Main` methods.
```

Comment:
Note for me, depending on how soon we merge this: I'm update the options section to use the new MSBuild properties rather than the `csc` spellings. I'll have to update his depending on the order of merging. See #22797 for the work in progress.